keps/sig-node/4753-expose-ownerref-downard-api/README.md

```diff
@@ -0,0 +1,592 @@
+This KEP proposes to expose the ownerReferences via `valueFrom` and downward API, so that the pod can directly pass its `ownerReferences` to the managed/created new object without having to call the API server.
+
+This KEP is intentionally scoped small, and intends to keep the scope just to `ownerReferences` and not to other fields of the owner object to prevent security risks.
+
+## Motivation
+
+Deployments and ReplicaSets can orphan and adopt pods dinamically. Any additional information we provide can change over time. On day 1 a pod named "foobar-deadbeef93-xyz76" is owned by replicaset "foobar-deadbeef93" which is owned by deployment "foobar", but after a node restart, the pod name could change to "foobar-deadbeef93-abc12", leaving the object created by "foobar-deadbeef93-xyz76" orphan, and triggering unwanted behavior. Enabling the pod to pass its ownerReferences to the new object it manages will prevent objects from being orphaned by pods that are owned by a higher level object.
+
+### Goals
+
+* Gain access to pod `ownerReferences` on pods through volume mounts.
+* Gain access to pod `ownerReferences` on pods through environmental variables.
+
+### Non-Goals
+
+- Not to expose additional pod info outside of `metadata.ownerReferences`.
+
+## Proposal
+
+The initial design includes:
+
+* Being able to pass pod `ownerReferences` to volumes with fieldPath of `metadata.ownerReferences`.
+* Being able to pass pod `ownerReferences` to environmental variables with fieldPath of `metadata.ownerReferences`.
+
+### User Stories (Optional)
+
+#### Story 1
+xref https://github.com/kubernetes-sigs/node-feature-discovery/issues/1752
+
+CustomResource object created by a pod owned by a DaemonSet is garbage collected when the pod is deleted by unexpected reasons. This triggers unwanted behavior in the system.
+If the CustomResource object inherits the ownerReferences from the pod, it will be garbage collected only when the DaemonSet is deleted. This will add extra reliability to the system by projecting the ownerReferences from the Deployment/DaemonSet/StatefulSet to the CustomResource object.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- Security Risk: Some environments may not want to expose the ownerReferences of a pod to the pod itself.
+  - Mitigation: This feature could be added as a feature gate, disabled by default.
+
+## Design Details
+
+The intention behind this design is not to add extra structs or fields
+but to extend the existing `downward API` to include the `metadata.ownerReferences` field.
+
+### Implementation Details
```

Comment:
Can you please say more about how this will present both as a file and in an env?

Can a user request a specific index of ownerRefs or does it always get all of them?  

What does the value of the env var actually look like?  OwnerRef is a struct with several fields, are we saying it will be JSON-formatted?  Or some other bespoke format?  What if I have 2 or 5 or 10 owners?  What if I have 0 owners.

Same question, but for file.  If I `cat` that file - what do I see?

Reply:
How about now?